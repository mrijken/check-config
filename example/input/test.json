{
    "list": [
        "to_be_removed",
        "to_be_kept"
    ],
    "to_be_removed": "foo",
    "to_be_kept": {
        "key": "value"
    },
    "regex": "matches"
}