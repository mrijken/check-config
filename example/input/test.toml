to_be_removed = "value"
regex = "matches"
list = ["to_be_removed", "to_be_kept"]

[to_be_kept]
key = "value"
