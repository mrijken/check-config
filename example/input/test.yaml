to_be_removed: value
to_be_kept:
  key: value
list:
  - to_be_kept
  - to_be_removed
regex: matches
