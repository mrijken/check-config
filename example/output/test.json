{
    "list": [
        "to_be_kept",
        "to_be_added"
    ],
    "regex": "matches",
    "to_be_added": "value",
    "to_be_kept": {
        "key": "value"
    }
}