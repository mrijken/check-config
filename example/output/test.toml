regex = "matches"
list = ["to_be_kept", "to_be_added"]
to_be_added = "value"

[to_be_kept]
key = "value"
