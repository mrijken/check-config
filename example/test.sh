SCRIPT_DIR=$(dirname "$BASH_SOURCE")
cd $SCRIPT_DIR
rm -r output
cp -r input output
cargo run -- --fix -p "$SCRIPT_DIR/checkers/checkers.toml" -vv
diff -w -B expected_output output
if [ $? -eq 0 ]; then
    echo "Expected output matches actual output"
else
    echo "Expected output does not match actual output"
fi
