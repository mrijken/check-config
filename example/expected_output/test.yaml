regex: matches
to_be_kept:
  key: value
list:
- to_be_kept
- to_be_added
to_be_added: value
